File: CS/Converters/TextUnderlineConverter.cs
using System;
using System.Windows;
using System.Windows.Data;
namespace DXRibbonNotepad.Converters {
    public class TextUnderlineConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            TextDecorationCollection coll = new TextDecorationCollection();
            if((bool)value)
                coll.Add(new TextDecoration());
            return coll;
        }
        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            throw new NotImplementedException();
        }
    }
}

File: VB/Converters/TextUnderlineConverter.vb
Imports System
Imports System.Windows
Imports System.Windows.Data

Namespace DXRibbonNotepad.Converters

    Public Class TextUnderlineConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.Convert
            Dim coll As TextDecorationCollection = New TextDecorationCollection()
            If CBool(value) Then coll.Add(New TextDecoration())
            Return coll
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace
